common/models/user.js

```diff
@@ -301,6 +301,25 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.app.models.AccessToken;
+
+    AccessToken.find({
+      where: {
+        userId: ctx.where.id,
+      },
+    }, function(err, tokens) {
+      tokens.forEach(function(token) {
+        AccessToken.destroyById(token.id, function() {
```

Comment:
I made a minor modification to my code, in defining the `AccessToken`.  It doesn't look like it will delete the user, unless I missed a step or did something wrong. I could have tried it with destroyAll if destroybyId is  not a good practice, but I changed the code to cope with the strategy above, but when u have the time please look at the code below. 

```
  User.observe('before delete', function(ctx, next) {
    var AccessToken = ctx.Model.relations.accessTokens.modelTo;

    AccessToken.find({
      where: {
        userId: ctx.where.id,
      },
    }, function(err, tokens) {
      tokens.forEach(function(token) {
        AccessToken.destroyById(token.id, function() {
          console.log('Deleted token %s for user ', token.id, token.userId);
        });
      });
    });

    next();
  });
```